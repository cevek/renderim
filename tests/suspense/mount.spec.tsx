import {createElement, Suspense} from 'renderim';
import {createLoadData, render} from '../base';

test('fast load', done => {
    const LoadData = createLoadData();
    const result = render(
        <Suspense timeout={50} fallback={<div>Loading...</div>}>
            <div class="wrapper">
                <LoadData ms={20} />
                Hello
            </div>
        </Suspense>,
    );
    expect(result.tree).toMatchSnapshot('1.root');
    setTimeout(() => {
        expect(result.getNextRestartedComponent()).toMatchSnapshot('2.Suspense result');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('3.Data');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('4.Suspense result');
        expect(result.getNextRestartedComponent()).toBeUndefined();
        done();
    }, 100);
});

test('zero timeout', done => {
    const LoadData = createLoadData();
    const result = render(
        <Suspense timeout={0} fallback={<div>Loading...</div>}>
            <div class="wrapper">
                <LoadData ms={20} />
                Hello
            </div>
        </Suspense>,
    );
    expect(result.tree).toMatchSnapshot('1.root');
    setTimeout(() => {
        expect(result.getNextRestartedComponent()).toMatchSnapshot('2.Suspense with loading');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('3.Suspense with loading2');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('4.Data');
        expect(result.getNextRestartedComponent()).toBeUndefined();
        done();
    }, 100);
});

test('slow load', done => {
    const LoadData = createLoadData();
    const result = render(
        <Suspense timeout={20} fallback={<div>Loading...</div>}>
            <div class="wrapper">
                <LoadData ms={50} />
                Hello
            </div>
        </Suspense>,
    );
    expect(result.tree).toMatchSnapshot('1.root');
    setTimeout(() => {
        expect(result.getNextRestartedComponent()).toMatchSnapshot('2.Suspense with loading');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('3.Suspense with loading2');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('4.Data');
        expect(result.getNextRestartedComponent()).toMatchSnapshot('5.Suspense result');
        expect(result.getNextRestartedComponent()).toBeUndefined();
        done();
    }, 100);
});
