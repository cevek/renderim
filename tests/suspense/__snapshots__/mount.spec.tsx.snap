// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fast load: 1.root 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`fast load: 2.Suspense result 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`fast load: 3.Data 1`] = `
"<Data>
   data
</Data>"
`;

exports[`fast load: 4.Suspense result 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data>
               data
            </Data>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`slow load: 1.root 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`slow load: 2.Suspense with loading 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`slow load: 3.Suspense with loading2 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            Loading...
         </div>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`slow load: 4.Data 1`] = `
"<Data>
   data
</Data>"
`;

exports[`slow load: 5.Suspense result 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data>
               data
            </Data>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`zero timeout: 1.root 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`zero timeout: 2.Suspense with loading 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            Loading...
         </div>
         <div>
            <Data/>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;

exports[`zero timeout: 3.Suspense with loading2 1`] = `
"<Data>
   data
</Data>"
`;

exports[`zero timeout: 4.Data 1`] = `
"<Suspense>
   <Boundary>
      <Fragment>
         <div>
            <Data>
               data
            </Data>
            Hello
         </div>
      </Fragment>
   </Boundary>
</Suspense>"
`;
